sdk/aqueduct/artifacts/bool_artifact.py

```diff
@@ -74,19 +74,19 @@ def get(self, parameters: Optional[Dict[str, Any]] = None) -> Union[bool, np.boo
                     "Parameterizing historical artifacts is not currently supported."
                 )
 
-        if parameters is None and self._get_content() is not None:
-            return self._get_content()
+        content = self._get_content()
+        if parameters is not None or content is None:
+            previewed_artifact = artifact_utils.preview_artifact(
+                self._dag, self._artifact_id, parameters
+            )
+            content = previewed_artifact._get_content()
+            assert isinstance(content, bool) or isinstance(content, np.bool_)
```

Comment:
Do we have to do this just for `self._set_content(content)` to not complain? Because it seems that we are doing the same assertion in line 89 before returning anyway.

Reply:
actually yeah, I initially wanted to put this here just so we wouldn't ever set the content of this artifact to a non-bool data type. However, that seems unnecessary now that I think about it - these assertions are really just for type-checking, since it content value should never be a bool.